client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/pattern-large-preview.tsx

```diff
@@ -224,7 +224,15 @@ const PatternLargePreview = ( {
 			}
 		};
 
-		scrollIntoView();
+		// Only scroll when the pattern is added via the pattern list panel.
+		// This prevents auto-scrolling when the pattern is added via shuffle, which causes the pattern action bar to jump around.
+		const focusedElement = document.activeElement;
+		if (
+			! focusedElement ||
+			focusedElement.classList.contains( 'pattern-list-renderer__pattern-list-item' )
```

Comment:
How about checking whether the focused element is inside the large preview? It has a chance we change the behavior somehow, e.g.: the element to add the pattern.

Reply:
I tried that but when section patterns are shuffled, `document.activeElement` becomes:
![Screenshot 2023-11-29 at 8 27 09 AM](https://github.com/Automattic/wp-calypso/assets/797888/209a17e3-50c6-4140-9774-14dba8e47f5a)

This doesn't happen when header/footer patterns are shuffled, I'm assuming it's because its DOM isn't re-rendered. This makes the approach "checking inside the large preview" not so straightforward, unfortunately 😕

Reply:
Ok 🥲